appnav/src/main/kotlin/io/element/android/appnav/di/DefaultSyncOrchestrator.kt

```diff
@@ -49,10 +48,10 @@ class DefaultSyncOrchestrator @AssistedInject constructor(
 
     private val initialSyncMutex = Mutex()
 
-    private var coroutineScope: CoroutineScope? = null
-
     private val tag = "SyncOrchestrator"
 
+    private val coroutineScope = CoroutineScope(baseCoroutineScope.coroutineContext + CoroutineName(tag) + dispatchers.io)
+
```

Comment:
I guess this scope is cancelled when the baseCoroutineScope is cancelled.

Have you tested what happen when the session is logged out?

It should work as expected, but maybe worth double checking.

Reply:
Good call, thanks to this we found a critical error. Patch incoming!